build/gulpfile.sql.js

```diff
@@ -130,3 +131,19 @@ function installSsmsMin() {
 gulp.task('install-ssmsmin', () => {
 	return installSsmsMin();
 });
+
+gulp.task('check-all-files', () => {
```

Comment:
it's probably better to send these as two separate PRs.  I think the CSS\agent changes seem pretty straightforward.  

I'm not sure about the files list change.  I am sure when we'd run this task and what we'd do with the result.  My assumption is that files will move around relatively often and that unless there is some associated bug this isn't a problem.  I'd personally rather focus on getting tests for the bugs that would result if files are moved around unexpectedly.

Reply:
The last failure we had was because of the missing file. Perhaps not all of the files, but we can look for certain important files for a sanity check. I'll put the css commit in another PR.

Reply:
Yeah I think we should have some more discussion. This seems like it's going to be a pain to maintain - especially if you're going to be included a huge amount of files. What's the plan for maintaining this? This overall just seems like overkill for what we need - and as Karl said if we can get it so our actual tests cover this kind of checking for us that's a much better solution I think.

There's really only a very few files that need to be in a certain place for external callers (such as the gulp tasks) so at the very least we should be trying to scope it down to just "important" files that fit that criteria.

---
In reply to: [325378108](https://github.com/microsoft/azuredatastudio/pull/7244#discussion_r325378108) [](ancestors = 325378108)

Reply:
This test wouldn't fail a lot but would be helpful in big merges from VSCode. I've also seen failures before because the packaged builds were missing image files/icons. I don't think we have any check for that right now. The best part is that packaging also wouldn't complain about it until you open the Console for the packaged app. So, a broken app could very well have a clean build.